ops/buildkite/deploy-content-preview

```diff
@@ -0,0 +1,163 @@
+#  update them on the staging server. You then just need to run `yarn
+#  deploy-content-preview` to deploy the updated functions code to the preview
+#  URL.
+#
+#  ## CREATING CLOUDFLARE PAGES API TOKEN
+#
+#  This script requires a Cloudflare API token to deploy to the preview URL.
+#
+#  In particular, `wrangler` uses CLOUDFLARE_ACCOUNT_ID and
+#  CLOUDFLARE_API_TOKEN, read from owid-grapher/.env on the staging server.
+#
+#  - CLOUDFLARE_ACCOUNT_ID: You can find your account ID at
+#  https://dash.cloudflare.com/ > Our World in Data. The account ID is in the
+#  URL, e.g. https://dash.cloudflare.com/1234567890abcdef1234567890abcdef
+#
+# - CLOUDFLARE_API_TOKEN: You can create your token at
+#  https://dash.cloudflare.com/profile/api-tokens. Use the "Create Custom Token"
+#  template, then give it the "Account > Cloudflare Pages > Edit" permission.
+#  Use a TTL corresponding to the duration of your test (e.g. 2 weeks).
+#
+#  Save both these values in owid-grapher/.env on the staging server (not
+#  locally).
+
+
+set -o errexit
+set -o pipefail
+set -o nounset
+
+GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+HOST="staging-site-$GIT_BRANCH"
+
+as_owid() {
+    ssh -t -o "LogLevel=QUIET" -o "StrictHostKeyChecking=no" "owid@$HOST" "$@"
+}
+
+# Cloudflare Pages settings
+PROJECT_NAME=owid
+
+# Set PREVIEW_BRANCH to the current git branch name if not set
+PREVIEW_BRANCH=${PREVIEW_BRANCH:-$GIT_BRANCH}
+
+# URL of Cloudflare page
+BAKED_BASE_URL="https://$PREVIEW_BRANCH.$PROJECT_NAME.pages.dev"
+
+deploy_site () {
+    echo "--- Deploy site to $BAKED_BASE_URL from $HOST"
+
+    create_dist
+    deploy_to_cloudflare
+
+    echo "--- Site deployed to $BAKED_BASE_URL from $HOST"
+}
+
+create_dist() {
+    echo '--- Creating dist/ folder'
+    # Define a list of excluded directories for rsync
+    EXCLUDES=(grapher/data/variables/ .git/ grapher/exports/ exports/ uploads/)
+
+    # Build rsync command with excluded directories
+    RSYNC_COMMAND=("rsync" "-havzq" "--delete")
+    for EXCLUDE in "${EXCLUDES[@]}"; do
+        RSYNC_COMMAND+=("--exclude=$EXCLUDE")
+    done
+
+    as_owid "${RSYNC_COMMAND[@]}" live-data/bakedSite/ dist/
+
+    # remove .etag of gdoc images from dist
+    as_owid rm dist/images/published/*.etag
+
+    # copy over Pages Functions as-is
+    as_owid rsync -havzq --delete owid-grapher/functions dist/
+
+    # we need node_modules for Pages Functions
+    as_owid rm -rf dist/node_modules
+    as_owid cp -rL owid-grapher/node_modules dist/ # the `L` flag copies over symlinked files, too, which we need for @ourworldindata/utils etc.
+    # hack to use the global `wrangler`` in deploy_to_cloudflare. `npx wrangler`
+    # otherwise uses the local version in node_modules, but fails to resolve
+    # some paths due to the copy.
+    as_owid rm -rf dist/node_modules/.bin/wrangler*  dist/node_modules/wrangler
+    as_owid cp owid-grapher/_routes.json dist/_routes.json
```

Comment:
@Marigold currently, `wrangler` is not in package.json, so `npx wrangler` installs it globally. With this PR stack, `wrangler` will be found in `node_modules` and used by `npx`. This breaks the `wrangler pages deploy` command below and I believe will break production deploys (./deploy-content) in the same way. Ideally, we would be able to use the local `node_modules` version, but I couldn't find a way to do so.

I'm not too comfortable running `deploy-content` without proper assessment given all the side effects, so I wanted to get your views on this first.

Reply:
I'm not sure I fully understood it, but I assume it would be way safer to install wrangler instead of running it with `npx`. Let me create a PR for it and test it out.

Reply:
I gave it a try and only calling the script directly worked with local version
```
node_modules/wrangler/bin/wrangler.js pages deploy 
```
I'm not sure if that's better or worse... what do you think @mlbrgl ?

Reply:
interesting, thanks! I somehow got so fixated on making it work through `npx` and didn't think of that. I would have slightly preferred being able to run it through `node_modules/.bin/wrangler` to isolate us a bit more from changes in wrangler's file structure but that gave me the same error as before

```
Error: Cannot find module '/home/owid/dist/node_modules/wrangler-dist/cli.js'
```
(it looks like a relative path is messed up somewhere, and an extra `../` is applied which makes the script jump out of `node_modules/wrangler`)

Reply:
Sorry, I was in a hurry yesterday, messed up testing, requested your review and then made a draft out of it 🤦

I've changed it to `node_modules/wrangler/bin/wrangler.js` and left a comment about `node_modules/.bin/wrangler`. It should work now.

Reply:
Have you tried running it via yarn?

Reply:
@danyx23 yes, but yarn needs a `package.json`, which is not there in `dist/` (see https://github.com/owid/owid-grapher/pull/3059#discussion_r1440589617)